server/service/topicService.js

```diff
@@ -489,7 +489,33 @@ exports.getActiveTopicEnrollmentsByUserAndTopicIdWithEverything = async function
     }
 };
 
+// Takes in a topicId and finds each resourceId associated with it.
+exports.getAllResourceIdsFromTopic = async function ( topicId ) {
 
+    let text = "SELECT * from topic_resource where topic_id = $1";
+    let values = [ topicId ];
+    let resourceIds = [];
+
+    try {
+
+        let res = await db.query( text, values );
+
+        // console.log( " RESPONSE : " + JSON.stringify( res ) );
+        if ( res.rowCount > 0 ){
+            for ( let i=0; i<res.rowCount; i++ ){
+                resourceIds[i] = res.rows[i].resource_id;
```

Comment:
Should this push onto the array? I'm not the most savvy in JS but I believe this will lead to an error because a null index is being addressed.

Reply:
Nope, JS will dynamically largen the array when needed.

Reply:
Okay thank you! I will approve these changes.